packages/squiggle-lang/src/dist/SampleSetDist/index.ts

```diff
@@ -93,7 +93,10 @@ export class SampleSetDist extends BaseDist {
     if (rightCutoff !== undefined) {
       truncated = truncated.filter((x) => x <= rightCutoff);
     }
-    return SampleSetDist.make(truncated);
+    return Result.bind(
+      SampleSetDist.make(truncated),
+      (dist) => SampleSetDist.make(dist.sampleN(this.samples.length)) // resample to original length
```

Comment:
Note that sampleN might take the same samples twice. There might be a way to better ensure this doesn't happen - though it's not a major issue.